quarkchain/cluster/shard_state.py

```diff
@@ -636,14 +646,6 @@ def __validate_block(
                 % (xshard_gas_limit, block.meta.evm_xshard_gas_limit)
             )
 
-        if (
```

Comment:
I think we still need to make sure blocks only contain whitelisted tx.

note that other miners can always manually craft a block including unwanted transactions.

Reply:
then the block would be rejected as transactions cannot be validated

Reply:
this is also covered by unittest:
https://github.com/QuarkChain/pyquarkchain/pull/475/files#diff-f0cb0611ad35e203a2117908d4a5a812R2241
even the block contains transactions (unwhitelisted), it would be rejected